src/transport/Tuple.h

```diff
@@ -0,0 +1,148 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ *
+ * Defines a compound transport type (tuple) that can merge several transports
+ * to look like a single one.
+ */
+#ifndef TRANSPORT_TUPLE_H_
+#define TRANSPORT_TUPLE_H_
+
+#include <tuple>
+#include <type_traits>
+
+#include <transport/Base.h>
+
+namespace chip {
+namespace Transport {
+
+/**
+ * Groups together several transports of different types and presents them as a unified transport.
+ *
+ * The usage intent is to be able to group several distinct transport types and make them look
```

Comment:
Is the intent of that grouping to do iterative unicast across them to effect a multicast of sorts or to only use one at any given time depending on some matching set of criteria of message against transport?

Reply:
It may evolve, however my current thought is that transports can say if a peer address can be reached via 'self'. This means:
  - BLE only supports BLE peer addresses
  - Multicast sounds as only supported by UDP
  - if multiple transports can be used (e.g. both TCP and UDP can reach something), then first one in the list wins

We have a 'PeerAddress' that encodes the peer and transports can say yes/now for being suitable for a given peer address. 

Design is difficult because we only have one transport as of yet (UDP). I see BLE is being worked on and we may also build TCP based on WML to have something better defined.

Reply:
I think some of that context you just shared would be valuable to memorialize in the template description.

Reply:
Added docs.